sat/cli/bootsys/mgmt_power.py

```diff
@@ -428,10 +428,71 @@ def do_power_on_ncns(args):
                         except FatalPlatformError as err:
                             LOGGER.error(f'Failed to unfreeze Ceph on storage NCNs: {err}')
                             sys.exit(1)
-                        LOGGER.info('Ceph freeze completed successfully on storage NCNs.')
+                        LOGGER.info('Ceph unfreeze completed successfully on storage NCNs.')
+
+                        # Mount Ceph and S3FS filesystems on ncn-m001
+                        try:
+                            mount_filesystems_on_ncn('ncn-m001')
+                        except MountError as err:
+                            LOGGER.error(f'Failed to mount filesystems on ncn-m001: {err}')
+                            sys.exit(1)
+                        LOGGER.info('Mounted Ceph and S3FS filesystems on ncn-m001 successfully.')
 
         except ConsoleLoggingError as err:
             LOGGER.error(f'Aborting boot of NCNs due failure to set up NCN console logging: {err}')
             raise SystemExit(1)
 
     LOGGER.info('Succeeded with {}.'.format(action_msg))
+
+
+def mount_filesystems_on_ncn(ncn):
+    """Mount Ceph and S3FS filesystems on a given NCN.
+
+    Args:
+        ncn: The hostname of the NCN where the filesystems will be mounted.
+
+    Raises:
+        MountError: If mounting any filesystem fails.
+    """
+    filesystems = {
+        "ceph": "awk '{ if ($3 == \"ceph\") { print $2; }}' /etc/fstab",
+        "fuse.s3fs": "awk '{ if ($3 == \"fuse.s3fs\") { print $2; }}' /etc/fstab"
+    }
+
+    for fs_type, awk_command in filesystems.items():
+        try:
+            result = subprocess.run(awk_command, shell=True, capture_output=True, text=True, check=True)
```

Comment:
Remember that `sat` commands are actually executed in the `cray-sat` container, and this container will not have the same `/etc/fstab` file contents as on the actual ncn-m001 host.

For example, here are the ceph and fuse.s3fs entries from the ncn-m001 host:

```
ncn-m001:~ # awk '{ if ($3 == "ceph") { print; } }' /etc/fstab
admin-tools@.admin-tools=/ /etc/cray/upgrade/csm ceph
ncn-m001:~ # awk '{ if ($3 == "fuse.s3fs") { print; } }' /etc/fstab
sds /var/opt/cray/sdu/collection-mount fuse.s3fs _netdev,allow_other,passwd_file=/root/.sds.s3fs,url=http://rgw-vip.nmn,use_path_request_style,use_cache=/var/lib/s3fs_cache,check_cache_dir_exist,use_xattr,uid=2370,gid=2370,umask=0007,allow_other 0 0
config-data /var/opt/cray/config-data fuse.s3fs _netdev,allow_other,passwd_file=/root/.config-data.s3fs,url=http://rgw-vip.nmn,use_path_request_style,use_xattr 0 0
```

Note that there are no such entries in the sat container:

```
ncn-m001:~ # sat bash
(57bb42e57531) sat-container:/sat/share # awk '{ if ($3 == "ceph") { print; } }' /etc/fstab
(57bb42e57531) sat-container:/sat/share # awk '{ if ($3 == "fuse.s3fs") { print; } }' /etc/fstab
(57bb42e57531) sat-container:/sat/share #
```

I think this means that your code is not actually going to perform any mounts. Instead, you should be using Paramiko (SSH) to execute commands on `ncn-m001`.